### PR TITLE
Usual filters

### PR DESCRIPTION
**Pull request by domi4484**
image

### CODE REVIEW

File: enhanced_relation_editor_widget/core/features_model_filter.py [hunk 1]
```diff
@@ -69,4 +144,58 @@ def filterAcceptsRow(self,
...
         return self._quick_filter in rowDisplayRole
 
+    def _prepare_filtered_features(self):
+
+        self._legacy_filter_filtered_features = list()
+
+        if not self._legacy_filter_expression.isValid():
+            return
+
+        distanceArea = QgsDistanceArea()
```
Comment (signedav) [inline enhanced_relation_editor_widget/core/features_model_filter.py:154]: `distance_area`?

File: enhanced_relation_editor_widget/core/features_model_filter.py [hunk 2]
```diff
@@ -69,4 +144,58 @@ def filterAcceptsRow(self,
...
+        distanceArea.setSourceCrs(self._layer.crs(), QgsProject.instance().transformContext())
+        distanceArea.setEllipsoid(QgsProject.instance().ellipsoid())
+
+        fetchGeom = self._legacy_filter_expression.needsGeometry()
```
Comment (signedav) [inline enhanced_relation_editor_widget/core/features_model_filter.py:158]: `fetch_geom`?

File: enhanced_relation_editor_widget/gui/feature_filter_widget.py [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+                            neww: QWidget):
+        self.mFilterLayout.removeWidget(oldw)
+        oldw.setVisible(False)
+        self.mFilterLayout.addWidget(neww, 0, 0)
+        neww.setVisible(True)
+        neww.setFocus()
+
+    def onFilterQueryTextChanged(self,
+                                 value: str):
+        self.mFilterQueryTimer.start(300)
```
Comment (signedav) [inline enhanced_relation_editor_widget/gui/feature_filter_widget.py:422]: I didn't checked every line of the refactored class...

File: enhanced_relation_editor_widget/core/features_model_filter.py [hunk 4]
```diff
@@ -1,21 +1,50 @@
...
+                 layer: QgsVectorLayer,
+                 canvas: QgsMapCanvas,
                  parent: QObject = None):
         super().__init__(parent)
 
+        self._layer = layer
+        self._canvas = canvas
         self._quick_filter = str()
         self._map_filter = list()
+        self._legacy_filter = FeaturesModelFilter.LegacyFilter.ShowAll
```
Comment (signedav) [inline enhanced_relation_editor_widget/core/features_model_filter.py:41]: Would not call em "legacy". I would prefer calling it "feature" filter. It's not that descriptive but you already know it like that. Or maybe "normal"... up to you. But legacy sounds bad for a new plugin :-)

**Review by signedav — APPROVED** [review]
Looks good.

File: enhanced_relation_editor_widget/core/features_model_filter.py (hunk 4 above)
Comment (domi4484) [reply enhanced_relation_editor_widget/core/features_model_filter.py:41]: Thanks for the suggestion!

File: enhanced_relation_editor_widget/core/features_model_filter.py (hunk 1 above)
Comment (domi4484) [reply enhanced_relation_editor_widget/core/features_model_filter.py:154]: this are code parts copied 1 to 1 from cpp... I will make a run all over the code and look to fix all those things

**Pull request merged**